ledger/src/blockstore_db.rs

```diff
@@ -1416,21 +1420,19 @@ fn get_cf_options<C: 'static + Column + ColumnName>(
     options.set_max_bytes_for_level_base(total_size_base);
     options.set_target_file_size_base(file_size_base);
 
-    // TransactionStatusIndex and ProgramCosts must be excluded from LedgerCleanupService's rocksdb
-    // compactions....
-    if matches!(access_type, AccessType::PrimaryOnly) && !excludes_from_compaction(C::NAME) {
+    let disable_auto_compactions = disable_auto_compactions(access_type);
+    if disable_auto_compactions {
+        options.set_disable_auto_compactions(true);
+    }
+
+    if !disable_auto_compactions && !exclude_column_from_compaction(C::NAME) {
         options.set_compaction_filter_factory(PurgedSlotFilterFactory::<C> {
             oldest_slot: oldest_slot.clone(),
             name: CString::new(format!("purged_slot_filter_factory({})", C::NAME)).unwrap(),
             _phantom: PhantomData::default(),
         });
     }
 
-    // Disable automatic compactions in maintenance mode to prevent accidental cleaning
-    if matches!(access_type, AccessType::PrimaryOnlyForMaintenance) {
```

Comment:
Just curious.  Do you happen to know the exact reason for this?

Reply:
I think the old comment may have been a little inaccurate.

We only use this mode for `solana-ledger-tool purge` command. With the `purge` command, the idea is we'd wipe a (possibly large) chunk of data and then want to perform a manual compaction to immediately reclaim that disk space.

The other PR that actually changes behavior for the modes improves upon this comment as well